interfaces/repo.go

```diff
@@ -265,6 +265,71 @@ func (r *Repository) RemoveSlot(snapName, slotName string) error {
 	return nil
 }
 
+// ResolveConnect resolves potentially missing plug or slot names and returns a
+// fully populated connection reference.
+func (r *Repository) ResolveConnect(plugSnapName, plugName, slotSnapName, slotName string) (*ConnRef, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	if plugSnapName == "" {
+		return nil, fmt.Errorf("cannot resolve connection, plug snap name is empty")
+	}
+	if plugName == "" {
+		return nil, fmt.Errorf("cannot resolve connection, plug name is empty")
+	}
+	// Ensure that such plug exists
+	plug := r.plugs[plugSnapName][plugName]
+	if plug == nil {
+		return nil, fmt.Errorf("cannot resolve connection, plug %q from snap %q, no such plug", plugName, plugSnapName)
+	}
+
+	if slotSnapName == "" {
+		// Use the core snap if the slot-side snap name is empty
+		switch {
+		case r.slots["core"] != nil:
+			slotSnapName = "core"
+		case r.slots["ubuntu-core"] != nil:
+			slotSnapName = "ubuntu-core"
+		default:
+			// XXX: perhaps this should not be an error and instead it should
+			// silently assume "core" now?
+			return nil, fmt.Errorf("cannot resolve connection, slot snap name is empty")
+		}
+	}
+	if slotName == "" {
+		// Find the unambiguous slot that satisfies plug requirements
+		var candidates []string
+		for candidateSlotName, candidateSlot := range r.slots[slotSnapName] {
+			// TODO: use some smarter matching (e.g. against $attrs)
+			if candidateSlot.Interface == plug.Interface {
+				candidates = append(candidates, candidateSlotName)
```

Comment:
I don't think a candidates list is really needed here. We are only interested in a single match, so we could just use a string for candidate and return out early with an error from the loop if we already have it set.